### DISCUSSION TITLE
Original text visible beneath translated text.

### DISCUSSION THREAD

**Discussion by chad130** [pymupdf/PyMuPDF#2317] [Looking for help] [answered]
![image](https://user-images.githubusercontent.com/116608057/229382692-6f2b5f72-7428-4ad8-885e-397489d7b4b5.png)
I'm having trouble figuring out how to identify and remove the original non-English text from a translated pdf. Is this possible? I can pull the English text using "get_text()", so maybe I have to construct a new pdf somehow?

This occurs after turning a non-searchable pdf to an image(pymupdf), then ocr(Tesseract), then back to pdf. I then run the pdf through Google Translate and the overwritten text is the result. 

pdf before translation:
[page_4.pdf](https://github.com/pymupdf/PyMuPDF/files/11133540/page_4.pdf)

pdf after translation:
[page_4_tr.pdf](https://github.com/pymupdf/PyMuPDF/files/11133541/page_4_tr.pdf)

**Comment by JorjMcKie** [comment, marked as answer]
The translation has converted the original page to an image, upon which the translated text has been written.
So the easiest way probably is to just remove that image:
```python
import fitz
doc=fitz.open("page_4_tr.pdf")
page=doc[0]
print(page.get_images())
[(13, 0, 1681, 2378, 8, 'DeviceRGB', '', 'FXX1', 'DCTDecode')]
page.delete_image(13)
doc.save("english.pdf", garbage=3, deflate=True)
```
This gives a page with only the English text - nothing else. Note the garbage collection and compression options.

**Reply by chad130** [reply]
Maybe I'm doing something wrong, but whenever I try this method, the "page.delete_image(13)" function triggers an AtrributeError:
AttributeError: 'Document' object has no attribute 'is_image'

Otherwise, I get the same response from get_images() as you. Knowing it's an image is new info, so I'll play around with that and maybe I'll find another way. Any additional help you could provide would be appreciated.

**Reply by JorjMcKie** [reply]
there was an error in an earlier release, which used a wrong method name
Either upgrade or do `Document.is_image = Document.xref_is_image`.

**Comment by chad130** [comment]
I think I'm using the latest version(1.21.1). I uninstalled, purged cache, and reinstalled to verify. The problem seems to be with the fitz.py file. It contains the old method you referenced.
![image](https://user-images.githubusercontent.com/116608057/229681244-67405d89-5daf-4a88-8064-28b5010d1221.png)

 Is it possible I'm not getting the latest version of fitz.py with my pip install? I receive version 4.0.2
![image](https://user-images.githubusercontent.com/116608057/229682028-63310291-7582-49ed-976e-8d7d0af8a93b.png)

Either way, the fix u suggested did the job. Alternatively I had success changing the xref_is_image method to is_image within the fitz.py.
